src/System.IO.Abstractions.TestingHelpers/MockPath.cs

```diff
@@ -115,13 +115,16 @@ public override string GetFullPath(string path)
             {
                 fullPath = DirectorySeparatorChar + fullPath;
             }
-            else if (isUnixRooted)
+            else if (isUnixRooted || isUnc)
```

Comment:
feels like we should only set isUnixRooted if "current directory is not a UNC path" which would resolve this issue as well. Although re-using the same slashes feels safest

Reply:
I agree, that sounds more correct.  I think `isUnixRooted` should only be `true` when the path starts with a forward slash.

Reply:
Done, much cleaner